backend/ASTU-backend-group-1/Infrastructure/AuthorizationMiddleware.go

```diff
@@ -0,0 +1,79 @@
+package infrastructure
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type AuthMiddleware struct {
+	collection *mongo.Collection
+}
+
+func NewAuthMiddleware(collection *mongo.Collection) GeneralAuthorizer {
+	return &AuthMiddleware{
+		collection: collection,
+	}
+}
+
+func (am *AuthMiddleware) AUTH(tokenString, secretKey string) jwt.Claims {
```

Comment:
Instead of using indexing for accessing the data in the claim u can create a customClaims struct and u can access the data from them as fields